llvm/lib/IR/Verifier.cpp

```diff
@@ -4224,6 +4224,33 @@ void Verifier::visitLoadInst(LoadInst &LI) {
   visitInstruction(LI);
 }
 
+static bool isConstantAddressSpace(unsigned AS) {
```

Comment:
constant address space is not an AMD only thing. I'm thinking probably we can split this function into target dependent (for example, to query whether an AS is a constant AS) and independent part.

Reply:
What would the independent part contain if we do not have a use case for it yet?